### PR TITLE
VAGOV-1637 Health Care Region Locations list page

### PR DESCRIPTION
**Pull request by aurora-a-k-a-lightning**
## Description
- Add the locations list for a health care region grouped by "main location" sorted by name alphabetically

## Testing done
- Locally with staging data

## Screenshots
screen shot 2019-02-21 at 11 29 17 am

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/drupal/metalsmith-drupal.js [hunk 1]
```diff
@@ -42,6 +42,29 @@ function writeDrupalIndexPage(files) {
...
+    // Keep these pages out of the sitemap until we remove
+    // the drupal prefix
+    private: true,
+  };
+}
+
+// Creates the top-level health care region list pages (Locations, Services, etc.)
+function createHealthCareRegionListPages(page, drupalPagePath, files) {
+  // Create the top-level locations page for Health Care Regions
+  files[`drupal${drupalPagePath}/locations/index.html`] = createFileObj(
```
Comment (jbalboni) [inline src/site/stages/build/drupal/metalsmith-drupal.js]: What's the reason behind adding `locations` to the path here? I'm a little hesitant to have url info embedded in our build scripts, instead of being sourced from Drupal.

File: src/applications/static-pages/facilities/FacilityListWidget.jsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      }
+
+      if (aName > bName) {
+        return 1;
+      }
+
+      return 0;
+    });
+
+    return sortedFacilities.map(facility => {
```
Comment (jbalboni) [inline src/applications/static-pages/facilities/FacilityListWidget.jsx]: I think I would return `sortedFacilities` from this function and move this map into the jsx in the render method. It's a little bit easier to read if the primary jsx output is in the render method of a component.

File: src/applications/static-pages/facilities/FacilityListWidget.jsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                >
+                  {facility.attributes.phone.mentalHealthClinic.replace(
+                    /[ ]?x/,
+                    '',
+                  )}
+                </a>
+              </div>
+            </div>
+            <div className="location-details-link">
+              <a href="#" className="usa-button usa-button-secondary">
```
Comment (jbalboni) [inline src/applications/static-pages/facilities/FacilityListWidget.jsx]: Is this going to have a link eventually?

File: src/applications/static-pages/facilities/FacilityListWidget.jsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+import { apiRequest } from '../../../platform/utilities/api';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import { buildAddressArray } from '../../facility-locator/utils/facilityAddress';
+
+export default class FacilityListWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      parsedFacilities: JSON.parse(this.props.facilities),
```
Comment (jbalboni) [inline src/applications/static-pages/facilities/FacilityListWidget.jsx]: I think I would parse this in the createFacilityListWidget file, before passing it into your React component. That creates a cleaner separation between the code that's dealing with the DOM and data stored in it, and your React code.

It'll also make it easier to unit test this component.

**Review by jbalboni — COMMENTED** [review]
I left a few comments, but this looks pretty good! We also usually write unit tests for components like FacilityListWidget, which you can dive into in this ticket or in a future one.

File: src/site/stages/build/drupal/metalsmith-drupal.js (hunk 1 above)
Comment (ahay-agile6) [reply src/site/stages/build/drupal/metalsmith-drupal.js]: i was trying to ask this in slack but i don't think the message went through! these listing pages for health care regions (Locations, health services, etc.) are reverse fields on the health care region page node, so, they do not have nodes themselves, nor do they have urls. only the detail pages do.
how should we proceed?

File: src/applications/static-pages/facilities/FacilityListWidget.jsx (hunk 3 above)
Comment (ahay-agile6) [reply src/applications/static-pages/facilities/FacilityListWidget.jsx]: yes, once we get the facility details pages

File: src/site/stages/build/drupal/metalsmith-drupal.js (hunk 1 above)
Comment (jbalboni) [reply src/site/stages/build/drupal/metalsmith-drupal.js]: I think this is ok, then. I'll think about maybe finding a better place to put this (especially if we end up having more like this), but this is fine for now.

File: src/platform/utilities/environment/index.js [hunk 5]
```diff
@@ -35,7 +35,7 @@ const ENVIRONMENT_CONFIGURATIONS = {
     BASE_URL: `http://${location.hostname}${
       location.port ? `:${location.port}` : ''
     }`,
-    API_URL: `http://${location.hostname}:3000`,
```
Comment (jbalboni) [inline src/platform/utilities/environment/index.js]: Need to revert this change

File: src/site/stages/build/drupal/graphql/GetAllPages.graphql.js [hunk 6]
```diff
@@ -11,12 +13,14 @@ module.exports = `
   ${fragments}
   ${landingPage}
   ${page}
+  ${healthCareRegionPage}
```
Comment (jbalboni) [inline src/site/stages/build/drupal/graphql/GetAllPages.graphql.js]: I think this needs to go in the other query files, too.

**Review by jbalboni — APPROVED** [review]
